src/main/java/uk/ac/ebi/ena/readtools/loader/common/producer/ProducibleDataChecker.java

```diff
@@ -8,16 +8,20 @@
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 */
-package uk.ac.ebi.ena.readtools.loader.common.feeder;
+package uk.ac.ebi.ena.readtools.loader.common.producer;
 
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * To be used together with {@link ProducibleData }. A method annotated with this is expected to check/validate the data
+ * held by the fields annotated with {@link ProducibleData }.
+ */
 @Retention( RetentionPolicy.RUNTIME )
 @Target( {ElementType.METHOD } )
 public @interface
-FeedableDataChecker
+ProducibleDataChecker

```

Comment:
Is it possible to replace ProducibleDataChecker annotation with a an interface with the required methods?